### ISSUE TITLE
Trouble polyfilling Promise for roadtrip

### ISSUE THREAD

**Issue by cmalven** [Rich-Harris/roadtrip#15]
Already said this, but I love roadtrip. It's great. Thank you.

I'm having some issues polyfilling Promises for roadtrip support in IE, however.

I'm using webpack to pull in roadtrip, so my file looks like so:

```js
import Promise from 'promise-polyfill';
import roadtrip from 'roadtrip';

if (!window.Promise) {
  window.Promise = Promise;
}

roadtrip.add('/', indexRoute).start();
```

Pretty simple. But in IE 11 will throw an error `Promise is undefined` related to `Promise: Promise` in the `roadtrip` object.

What I _think_ is happening is that `roadtrip` instantiates itself immediately upon being imported vs. only after being used, and because of the way webpack works all code pulled in with `import` will exist at the very beginning of the output js,  so as soon as I import `roadtrip` it immediately tries to find `window.Promise` and fails. So if it's true that road trip's search for `Promise` will happen before I even have the chance to execute any code (and polyfill `window.Promise`) how can I actually polyfill Promise for roadtrip?

Thanks for any hints you can provide.

**Comment by marianoviola** [comment]
To make it working you just need to import the polyfill in this way:

```js
import 'html5-history-api';
import 'promise-polyfill/src/polyfill';
import roadtrip from 'roadtrip';
```

Additionally, you should need to polyfill the [HTML5 History API](https://github.com/devote/HTML5-History-API).

**Comment by cmalven** [comment]
Thanks @marianoviola! This works, and it makes sense why it works. Importing in this way immediately instantiates the polyfill vs my method in which the polyfill isn't applied until runtime.

That being said, I would argue that roadtrip should wait to check for the existence of promises until roadtrip is actually used, reason being that with the current method there doesn't seem to be any possible way for me to "opt out" of roadtrip for browsers that don't support promises. Roadtrip is _always_ going to check for the existence of `window.Promise` (and possibly blow up) regardless of whether I actually end up using it or not.

But that's a separate issue, and your response addresses my original one. Thanks.
